upload/upload_test.go

```diff
@@ -0,0 +1,173 @@
+package upload
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestSuite struct {
+	suite.Suite
+}
+
+func TestConfigTestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (suite *TestSuite) SetupTest() {
+
+}
+
+func (suite *TestSuite) TestConfigNoFile() {
+
+	configPath := "nofile.conf"
+
+	_, err := loadConfigFile(configPath)
+	assert.EqualError(suite.T(), err, "open nofile.conf: no such file or directory")
+}
+
+func (suite *TestSuite) TestConfigWrongFile() {
+	var confFile = `
+access_token = eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDM0ODQxNTIsImlzcyI6Imh0dHBzOi8vbG9naW4udGVzdC5lZ2EubmJpcy5zZSIsInN1YiI6ImVnYS1ib3gtODkwMiJ9.JS-R4GUODAzbYzYs1hJR_92D9I3blfcGu5_yrK75L6QnIK6Aq6Po6kBbngJa-N9SRIzETxy5xd5uZthHaOQpFQ
+host_base = inbox.test.ega.nbis.se
+encoding = UTF-8
+secret_key = ega-box-8902
+host_bucket = inbox.test.ega.nbis.se
+multipart_chunk_size_mb = 50
+access_key = ega-box-8902
+use_https = True
+check_ssl_certificate = False
+check_ssl_hostname = False
+socket_timeout = 30
+human_readable_sizes = True
+guess_mime_type!True
+encrypt = False
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
+	assert.EqualError(suite.T(), err, "key-value delimiter not found: guess_mime_type!True\n")
+}
+
+func (suite *TestSuite) TestConfigMissingCredentials() {
+	var confFile = `
+access_token = eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDM0ODQxNTIsImlzcyI6Imh0dHBzOi8vbG9naW4udGVzdC5lZ2EubmJpcy5zZSIsInN1YiI6ImVnYS1ib3gtODkwMiJ9.JS-R4GUODAzbYzYs1hJR_92D9I3blfcGu5_yrK75L6QnIK6Aq6Po6kBbngJa-N9SRIzETxy5xd5uZthHaOQpFQ
+host_base = inbox.test.ega.nbis.se
+encoding = UTF-8
+host_bucket = inbox.test.ega.nbis.se
+multipart_chunk_size_mb = 50
+access_key = ega-box-8902
+use_https = True
+check_ssl_certificate = False
+check_ssl_hostname = False
+socket_timeout = 30
+human_readable_sizes = True
+guess_mime_type = True
+encrypt = False
+`
+	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(configPath.Name())
+
+	err = ioutil.WriteFile(configPath.Name(), []byte(confFile), 0644)
+	if err != nil {
+		log.Fatalf("failed to write temp config file, %v", err)
+	}
+
+	_, err = loadConfigFile(configPath.Name())
```

Comment:
```suggestion
	configPath, err := ioutil.TempFile(os.TempDir(), "s3cmd-")
	if err != nil {
		log.Fatal(err)
	}

	_, err = loadConfigFile(configPath.Name())
```

Reply:
No need for the write delete dance, just create an empty file.

Reply:
oh yes you are right on here, no need to actually write something in the file